### PR TITLE
[bugpoint] Remove block params

### LINKED ISSUE #1142: [bugpoint] Remove ebb params
```
    jump ebb1(v0)

ebb1(v0: i64):
```

==>

```
    jump ebb1

ebb1:
    v0 = iconst.i64 0
```

### PR DESCRIPTION
**Pull request by teapotd**
This PR adds a mutator for bugpoint that attempts to replace block params with constants. Fixes #1142.

### CODE REVIEW

File: cranelift/src/bugpoint.rs [hunk 1]
```diff
@@ -681,6 +752,13 @@ fn inst_count(func: &Function) -> usize {
         .sum()
 }
 
+fn block_params_count(func: &Function) -> usize {
```
Comment (bjorn3) [inline cranelift/src/bugpoint.rs:755]: Please inline this into the only call site.

File: cranelift/tests/bugpoint_test_expected.clif [hunk 2]
```diff
@@ -73,4 +42,12 @@ block117:
     v987 = iconst.i64 0
     call fn0(v0, v105, v1052, v883, v829, v987, v951, v842)
     trap user0
+
+block120:
+    v1052 = iconst.i16 0
```
Comment (bjorn3) [inline cranelift/tests/bugpoint_test_expected.clif:47]: Maybe we should try to move instructions to the entry block in a new mutator. I opened #1540 for this.

**Comment by bjorn3** [comment]
CI ran out of disk space?

**Comment by teapotd** [comment]
@bjorn3
> I just noticed that `Mutator::did_crash` is never called. Maybe you can remove it in this PR? Otherwise I will open a PR myself.

It is called in `reduce` and it's used by `MergeBlocks` mutator:

https://github.com/bytecodealliance/wasmtime/blob/7d88384c0fe087cd80a3d8ae1bbc80950b2b7d91/cranelift/src/bugpoint.rs#L764-L765

https://github.com/bytecodealliance/wasmtime/blob/7d88384c0fe087cd80a3d8ae1bbc80950b2b7d91/cranelift/src/bugpoint.rs#L650-L652

**Comment by bjorn3** [comment]
> It is called in reduce and it's used by MergeBlocks mutator:

Somehow the call didn't show up when I did Ctrl-F.

File: cranelift/src/bugpoint.rs [hunk 3]
```diff
@@ -652,6 +710,22 @@ impl Mutator for MergeBlocks {
...
+    if ty == F32 {
+        builder.f32const(0.0);
+        "f32const"
+    } else if ty == F64 {
+        builder.f64const(0.0);
+        "f64const"
+    } else {
+        builder.iconst(ty, 0);
+        "iconst"
+    }
```
Comment (abrown) [inline cranelift/src/bugpoint.rs:726]: Don't we need a `bconst` for the boolean types?

Comment (teapotd) [reply cranelift/src/bugpoint.rs:726]: I just moved this function from `ReplaceInstWithConst`, but yeah, we should add it.

Comment (abrown) [reply cranelift/src/bugpoint.rs:729]: Ok: https://github.com/bytecodealliance/wasmtime/issues/1633

**Pull request merged**